docs/DeveloperGuide.md

```diff
@@ -1,38 +1,337 @@
+
+`Command` instances are generated by their corresponding `Parser` classes (e.g. `AddCommand` is constructed by `AddParser`) and executed by `Main`.
+
+<br>
+
+### 5.5. Storage Component
+
+The `Storage` component is responsible for the saving and loading of program data from and to its data files. The following class diagram illustrates the structure of this component:
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Storage.puml)
+
+Storage is an interface supporting write/read data to/from computer storage:
+* Storage interface is implemented by JsonStorage in Mod Happy, which will read and load data to and from json format.
+* ListStorage can save a ArrayList of any class that extends Object in json format, and read them back into corresponding objects. (E.g. ModuleListStorage, TaskListStorage inherit from ListStorage)
+* There are navigability to Storage from Main and SaveCommand, which handles the load and write data to/from disk respectively.
+  <br><br><br>
+
+## 6. Implementation
+
+This section describes some details on how some features are implemented.
+<br>
+
+### 6.1. Edit Feature
+
+The edit feature allows the user to change a parameter of a task/module. The parameters of a module is its module description while the parameters of a task are its task name, task description and estimated working time. 
+
+The following sequence diagram illustrates the process:
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/EditSeqDiagrams/Edit.puml)
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/EditSeqDiagrams/GetModule.puml)
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/EditSeqDiagrams/GetTask.puml)
+
+Here is an example of editing the description of a task (First task of the module CS2113T):
+
+1. User inputs `edit task 1 -m CS2113T -d "Changed"`.
+2. `ModHappyParser` identifies the command word as `edit` and passes `task 1 -m CS2113T -d "Changed"` to `EditParser`.
+3. `EditParser` instantiates a `EditCommand` with `taskModule = "CS2113T"`, `taskIndex = 0`, `description = "Changed"`, `workingTime = null`, `taskname = null`. This is returned to `Main`.
+4. `Main` calls the `execute()` method of the `EditCommand` instance.
+5. `EditCommand` first gets the relevant `Module` and invokes `editTaskFromModule(targetModule)`.
+6. `editTaskFromModule(targetModule)` retrieves the task `targetTask` specified by the index and invokes `targetTask.setTaskDescription(description)` to change the description.
```

Comment:
Consider using object diagrams for each step to help readers better visualize the process (same applies for subsequent sections of DG). You may refer to this part of AddressBook's DG below 
![image](https://user-images.githubusercontent.com/69446722/161182988-503b933b-5629-46ce-99ae-2d31657ffaee.png)